server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/aggregators/SumValueAggregator.java

```diff
@@ -19,42 +19,45 @@
  */
 public class SumValueAggregator implements ValueAggregator<Double> {
 
-    public static final StarTreeNumericType STAR_TREE_NUMERIC_TYPE = StarTreeNumericType.DOUBLE;
+    public static final StarTreeNumericType VALUE_AGGREGATOR_TYPE = StarTreeNumericType.DOUBLE;
+    private CompensatedSum kahanSummation = new CompensatedSum(0, 0);
 
     @Override
     public MetricStat getAggregationType() {
         return MetricStat.SUM;
     }
 
     @Override
-    public StarTreeNumericType getStarTreeNumericType() {
-        return STAR_TREE_NUMERIC_TYPE;
+    public StarTreeNumericType getAggregatedValueType() {
+        return VALUE_AGGREGATOR_TYPE;
     }
 
     @Override
-    public Double getInitialAggregatedValue(Long segmentDocValue, StarTreeNumericType starTreeNumericType) {
-        return starTreeNumericType.getDoubleValue(segmentDocValue);
+    public Double getInitialAggregatedValueForSegmentDocValue(Long segmentDocValue, StarTreeNumericType starTreeNumericType) {
+        kahanSummation.reset(0,0);
+        kahanSummation.add(starTreeNumericType.getDoubleValue(segmentDocValue));
+        return kahanSummation.value();
     }
 
     @Override
-    public Double applySegmentRawValue(Double value, Long segmentDocValue, StarTreeNumericType starTreeNumericType) {
-        CompensatedSum kahanSummation = new CompensatedSum(0, 0);
-        kahanSummation.add(value);
+    public Double mergeAggregatedValueAndSegmentValue(Double value, Long segmentDocValue, StarTreeNumericType starTreeNumericType) {
+        assert kahanSummation.value() == value;
         kahanSummation.add(starTreeNumericType.getDoubleValue(segmentDocValue));
```

Comment:
Where is compensation ? that's the key part of algorithm right, can we check sum aggregator for example and fix the algo.
```
double sum = sums.get(bucket);
double compensation = compensations.get(bucket);
kahanSummation.reset(sum, compensation);
```

Reply:
makes sense, misread the implementation. I have accounted for sum and compensation in the calculation now.